pages/_app/index.jsx

```diff
@@ -110,17 +110,22 @@ class App extends NextApp {
         })
 
         // Subscribe store to router after initialization
-        router.events.on('routeChangeStart', this.handleRouteChange)
+        // We use routeChangeComplete otherwise router.query contains previous
+        // values
+        router.events.on('routeChangeComplete', this.handleRouteChange)
 
-        // Trigger App render
+        store.changeDataProvider(dataProviderId)
         this.setState({ isAuthorized: true })
       })
-      .catch((error) => {
+      .catch(async (error) => {
```

Comment:
I still don't like it async, let see what we can do 🤔

Reply:
You are very old school guy you know that? 😄 I surely noticed you prefer .then .catch instead of async/await and was waiting for this comment to come.